RoboJackets/robocup-software
# Striker tactic

## Description
(Premature) striker tactic (+ assist tactic).

Current function:
Pass -> Receive -> Shoot OR Capture -> Shoot depends on the situation

Targeted point for shoot is selected based on cost function
The striker is selected based on the probability that robot can score the goal

TODO:
need to find better cost func

## Steps to test

1. run sim
2. place ball and robots whatever you like on the opponent's side field
![image](https://user-images.githubusercontent.com/53668296/122654916-85dbdc00-d11c-11eb-8deb-892b57e64bdc.png)
something like this
3. Robot that has the highest probability to score the goal receive the ball or capture the ball itself and shoot.

Expected result: Goal!

## Concerns
Behavior is a bit unstable
